ecs/cluster.yaml

```diff
@@ -1335,9 +1335,7 @@ Resources:
       Code:
         ZipFile: |
           const lib = require('/opt/lib.js');
-          exports.handler = async function(event, context) {
-            return await lib.handler(event, context);
-          };
+          exports.handler = (event, context) => lib.handler(event, context);
```

Comment:
have you tested this? I think it should be `exports.handler = async (event, context) => lib.handler(event, context);` Otherwise you call an async function from a normal function

Reply:
Have I tested this specifically -- no, but it's the same pattern I use daily.  Adding the `async` flag introduces another Promise object that wraps another Promise.  It works, but it doesn't add functionality.

The only functional difference is what appears in the stack trace.  In the original, every stack trace would still contain a reference to `index.js`, whereas in my change all stack traces would stop at `lib.js`.  That's why my commit-comment about keeping the stack frame around.

I would certainly recommend picking either your original implementation or mine.  The code in your comment would work _only if_ AWS-Lambda's caller implementation is using `await`.  This is because `await` resolves Promise's recursively, while `.then()` _does not_ always recurse!  So if Lambda's code calls Promise.then() on the return value, the real Promise might be thrown away.

Reply:
I see.